traefik/templates/rbac/ingressclass-clusterrole.yaml

```diff
@@ -0,0 +1,22 @@
+{{- if and .Values.rbac.enabled .Values.rbac.namespaced }}
```

Comment:
Im wondering why we need a cluster role if the rbac is namespaced?

Reply:
You made me doubt: just de-deployed with namespace-scoped RBAC, reverting my change, and instead adding the IngressClass to the Role:

```
# kubectl get -o yaml role -n helm-traefik
apiVersion: v1
items:
- apiVersion: rbac.authorization.k8s.io/v1
  kind: Role
  metadata:
    name: traefik
    namespace: helm-traefik
...
  - apiGroups:
    - extensions
    - networking.k8s.io
    resources:
    - ingresses
    - ingressclasses
    verbs:
    - get
    - list
    - watch
  - apiGroups:
    - extensions
    - networking.k8s.io
    resources:
    - ingresses/status
...
```

I can confirm, such Role/RoleBinding won't do. During startup, Traefik logs the following:

```
E0303 11:10:05.742924       1 reflector.go:127] pkg/mod/k8s.io/client-go@v0.19.2/tools/cache/reflector.go:156: Failed to watch *v1beta1.IngressClass: failed to list *v1beta1.IngressClass: ingressclasses.networking.k8s.io is forbidden: User "system:serviceaccount:helm-traefik:traefik" cannot list resource "ingressclasses" in API group "networking.k8s.io" at the cluster scope
```

Accessing cluster scoped objects, you need a ClusterRole + ClusterRoleBinding, as there is no way for the API to limit your actions based on a Namespace.

Reply:
Now, TBH, using namespace scoped RBAC, do we actually need to get/list/watch IngressClass?
Are there any chance for a cluster that comes with two namespace-scoped ingress controllers watching over Ingresses in the same namespace, which could actually benefit from IngressClass?
In theory, it could. Then again, that sounds a bit far fetched/quite an edge case. We might just ignore that error, as no one's been complaining about this so far.